src/MenuActiveEffects.cpp

```diff
@@ -74,15 +74,67 @@ void MenuActiveEffects::loadGraphics() {
 void MenuActiveEffects::logic() {
 	effect_icons.clear();
 
+	bool alreadyOn;
+
 	for (size_t i = 0; i < stats->effects.effect_list.size(); ++i) {
 		if (stats->effects.effect_list[i].icon == -1)
 			continue;
 
 		const Effect &ed = stats->effects.effect_list[i];
+		
+		if(ed.group_stack){
+			alreadyOn = false;
+			for(size_t j=0; j < effect_icons.size(); ++j){
+				if(effect_icons[j].type == ed.type && effect_icons[j].name.compare(ed.name) == 0){
```

Comment:
You should compare the strings here as so: `effect_icons[j].name == ed.name`

It's consistent with how we do string comparisons across the code base and is easier to read. The relational string operators call `compare()` in the background, so it's purely an aesthetic difference AFAIK.